### PR TITLE
jormungandr: attribut 'templated' deleted from args

### PR DESCRIPTION
**Pull request by kadhikari**
Before calling the function "create_external_link" with parameter "templated", we should delete this parameter in the parameter **kwargs if present.

### CODE REVIEW

File: source/jormungandr/jormungandr/interfaces/v1/Schedules.py [hunk 1]
```diff
@@ -342,6 +342,8 @@ def wrapper(*args, **kwargs):
             delta = timedelta(seconds=1)
             dt = datetime.strptime(min_dt, "%Y%m%dT%H%M%S")
             if g.stat_interpreted_parameters.get('data_freshness') != 'realtime':
+                if "templated" in kwargs_links:
+                    del kwargs_links["kwargs"]
```
Comment (pbougue) [inline source/jormungandr/jormungandr/interfaces/v1/Schedules.py]: you probably meant: del kwargs_links["`templated`"]

**Review by pbougue — COMMENTED** [review]
I'd like some explainations on why it happens and why this is the clean way of doing it, because I don't understand much of it :)

**Comment by kadhikari** [comment]
@pbougue when "create_external_link" is called with the parameter "templated" and with presence of attribut "templated": ".." in kwargs, we have an error : "exceptions:TypeError: create_external_link() got multiple values for keyword argument 'templated' "

When we add &templated=true/false in the url navitia, the same error is raised.

File: source/jormungandr/jormungandr/interfaces/v1/serializer/api.py [hunk 2]
```diff
@@ -212,6 +212,8 @@ def get_links(self, obj):
                 else:
                      args[e.key] = e.values[0]
 
+            if "templated" in args:
```
Comment (xlqian) [inline source/jormungandr/jormungandr/interfaces/v1/serializer/api.py]: how about doing `args["templated"] = value.is_templated`, then remove `templated` at https://github.com/CanalTP/navitia/pull/2231/files#diff-4208927b849d6bd230bd4d6e04d9aa6cR220 ?

File: source/jormungandr/jormungandr/interfaces/v1/make_links.py [hunk 3]
```diff
@@ -225,9 +227,11 @@ def wrapper(*args, **kwargs):
             if isinstance(data, dict) or isinstance(data, OrderedDict):
                 data = self.prepare_objetcs(objects, True)
                 kwargs = self.prepare_kwargs(kwargs, data)
+                if "templated" in kwargs:
+                    del kwargs["kwargs"]
```
Comment (xlqian) [inline source/jormungandr/jormungandr/interfaces/v1/make_links.py]: you can just overwrite `templated` and remove it from the arguments

File: source/jormungandr/jormungandr/interfaces/v1/Journeys.py [hunk 4]
```diff
@@ -328,8 +328,10 @@ def wrapper(*args, **kwargs):
                     args['is_journey_schedules'] = True
                     allowed_ids.update(args.get('allowed_id[]', []))
                     args['allowed_id[]'] = list(allowed_ids)
+                    args['_type'] = 'journeys'
+
```
Comment (pbougue) [inline source/jormungandr/jormungandr/interfaces/v1/Journeys.py]: extra blank line + why don't you do the same for `rel`? (this stands for the whole PR)

**Review by pbougue — APPROVED** [review]
Looks pretty good that way (let's discuss about `rel`)

File: source/jormungandr/jormungandr/interfaces/v1/Journeys.py (hunk 4 above)
Comment (kadhikari) [reply source/jormungandr/jormungandr/interfaces/v1/Journeys.py]: because rel is not an optional parameter

**Comment by kadhikari** [comment]
jenkins test this please

File: source/jormungandr/jormungandr/interfaces/v1/make_links.py [hunk 5]
```diff
@@ -36,7 +36,7 @@
 from flask.ext.restful.utils import unpack
 
 
-def create_external_link(url, rel, _type=None, templated=False, description=None, **kwargs):
+def create_external_link(url, rel=None, _type=None, templated=False, description=None, **kwargs):
```
Comment (pbougue) [inline source/jormungandr/jormungandr/interfaces/v1/make_links.py]: most probably, you can remove that None

**Review by pbougue — APPROVED** [review]

**Pull request merged**